src/Oscoin/Configuration.hs

```diff
@@ -23,64 +30,93 @@ module Oscoin.Configuration
 
 import           Oscoin.Prelude hiding (option, (<.>))
 
+import           Data.Barbie
+import qualified Formatting as F
 import           Options.Applicative
 import           Paths_oscoin (getDataDir)
+import           System.Console.Option (Opt(Opt))
 import           System.Directory (XdgDirectory(..), getXdgDirectory)
 import           System.FilePath ((<.>), (</>))
 
 data ConfigPaths = ConfigPaths
     { xdgConfigHome :: FilePath
     , xdgConfigData :: FilePath
     , dataDir       :: FilePath
-    }
+    } deriving Show
 
 getConfigPaths :: IO ConfigPaths
 getConfigPaths = ConfigPaths
     <$> getXdgDirectory XdgConfig "oscoin"
     <*> getXdgDirectory XdgData   "oscoin"
     <*> getDataDir
 
-data Paths = Paths
-    { keysDir        :: FilePath
+data Paths f = Paths
+    { keysDir        :: f FilePath
     -- ^ Directory where the keypair is stored.
     --
     -- Default: $XDG_CONFIG_HOME/oscoin
-    , blockstorePath :: FilePath
+    , blockstorePath :: f FilePath
     -- ^ Path to the block store SQLite .db file.
     --
     -- Default: $XDG_DATA_HOME/oscoin/blockstore.db
-    , genesisPath    :: FilePath
+    , genesisPath    :: f FilePath
     -- ^ Path to the genesis YAML file.
     --
     -- Default: $oscoin_datadir/data/genesis.yaml
+    } deriving ( Generic
+               , FunctorB
+               , TraversableB
+               , ProductB
+               , ConstraintsB
+               , ProductBC
+               )
+
+deriving instance AllBF Show f Paths => Show (Paths f)
```

Comment:
I would imagine that writing such parsers via `barbies` (!!) is going to be (in practice) a mechanical process, but having a few documentation lines about how one should approach writing a new CLI `Parser` wouldn't harm, here.